File: orchestrator-java-reactive/src/main/java/com/github/mzr/coder/orchestrator/java/reactive/service/DecisionService.java
package com.github.mzr.coder.orchestrator.java.reactive.service;

import com.github.mzr.coder.model.ServiceResponse;
import reactor.core.publisher.Mono;

import java.util.List;

public interface DecisionService {
    Mono<List<String>> getStrategy(String name);
    Mono<ServiceResponse> getDecision(String name, List<ServiceResponse> serviceResponses);
}

File: orchestrator-kotlin-coroutine/src/main/java/com/github/mzr/coder/orchestrator/kotlin/reactive/service/DecisionService.kt
package com.github.mzr.coder.orchestrator.kotlin.reactive.service

import com.github.mzr.coder.model.ServiceResponse
import reactor.core.publisher.Mono

interface DecisionService {
    suspend fun getStrategy(name: String): List<String>
    suspend fun getDecision(name: String, serviceResponses: List<ServiceResponse>): ServiceResponse
}